src/Components/Server/test/Circuits/ComponentHubTest.cs

```diff
@@ -0,0 +1,209 @@
+                circuitIdFactory);
+            var serializer = new TestServerComponentDeserializer();
+            var circuitHandleRegistry = new TestCircuitHandleRegistry();
+            var hub = new ComponentHub(
+                serializer: serializer,
+                dataProtectionProvider: ephemeralDataProtectionProvider,
+                circuitFactory: circuitFactory,
+                circuitIdFactory: circuitIdFactory,
+                circuitRegistry: circuitRegistry,
+                circuitHandleRegistry: circuitHandleRegistry,
+                logger: NullLogger<ComponentHub>.Instance);
+
+            // Here we mock out elements of the Hub that are typically configured
+            // by SignalR as clients connect to the hub.
+            var mockCaller = new Mock<IHubCallerClients>();
+            var mockClientProxy = new Mock<IClientProxy>();
+            mockCaller.Setup(x => x.Caller).Returns(mockClientProxy.Object);
+            hub.Clients = mockCaller.Object;
+            var mockContext = new Mock<HubCallerContext>();
+            mockContext.Setup(x => x.ConnectionId).Returns("123");
+            hub.Context = mockContext.Object;
+
+            return (mockClientProxy, hub);
+        }
+
+        private class TestCircuitHandleRegistry : ICircuitHandleRegistry
+        {
+            private bool circuitSet = false;
+
+            public CircuitHandle GetCircuitHandle(object circuitKey)
+            {
+                return null;
+            }
+
+            public CircuitHost GetCircuit(object circuitKey)
+            {
+                if (circuitSet)
+                {
+                    var serviceScope = new Mock<IServiceScope>();
+                    var circuitHost = TestCircuitHost.Create(
+                        serviceScope: new AsyncServiceScope(serviceScope.Object));
+                    return circuitHost;
+                }
+                return null;
+            }
+
+            public void SetCircuit(object circuitKey, CircuitHost circuitHost)
+            {
+                circuitSet = true;
+                return;
+            }
+        }
+
+        private class TestServerComponentDeserializer : IServerComponentDeserializer
+        {
+            public bool TryDeserializeComponentDescriptorCollection(string serializedComponentRecords, out List<ComponentDescriptor> descriptors)
+            {
+                descriptors = default;
+                return true;
+            }
+        }
+
+        private class TestCircuitFactory : ICircuitFactory
+        {
+            public TestCircuitFactory(
+            IServiceScopeFactory scopeFactory,
+            ILoggerFactory loggerFactory,
+            CircuitIdFactory circuitIdFactory,
+            IOptions<CircuitOptions> options) { }
+
+            // We override the default `CreateCircuitHostAsync` so we can mock around
+            // the work of initializing a host and all its service dependencies.
```

Comment:
```suggestion
            // We override the default `CreateCircuitHostAsync` so we can mock around
            // the work of initializing a host and all its service dependencies.
```

I think the comment is obsolete now we're not overriding a virtual method.